modules/secrets/main.tf

```diff
@@ -0,0 +1,77 @@
+##############################################################################
+# Secret Group
+##############################################################################
+
+locals {
+  secret_groups = flatten([
+    for secret_group in var.secrets :
+    secret_group.existing_secret_group || secret_group.secret_group_name == null ? [] : [{
+      secret_group_name        = secret_group.secret_group_name
+      secret_group_description = secret_group.secret_group_description
+    }] if secret_group.secret_group_name != null
+  ])
+}
+
+data "ibm_sm_secret_groups" "existing_secret_groups" {
+  instance_id   = var.existing_sm_instance_guid
+  region        = var.existing_sm_instance_region
+  endpoint_type = var.endpoint_type
+}
+
+module "secret_groups" {
+  for_each                 = { for obj in local.secret_groups : obj.secret_group_name => obj }
+  source                   = "terraform-ibm-modules/secrets-manager-secret-group/ibm"
+  version                  = "1.2.2"
+  region                   = var.existing_sm_instance_region
+  secrets_manager_guid     = var.existing_sm_instance_guid
+  secret_group_name        = each.value.secret_group_name
+  secret_group_description = each.value.secret_group_description
+  endpoint_type            = var.endpoint_type
+}
+
+##############################################################################
+# Secrets
+##############################################################################
+
+locals {
+  secrets = flatten([
+    for secret_group in var.secrets :
+    secret_group.existing_secret_group ? [
+      for secret in secret_group.secrets != null ? secret_group.secrets : [] : merge({
+        secret_group_id   = secret_group.secret_group_name != null ? data.ibm_sm_secret_groups.existing_secret_groups.secret_groups[index(data.ibm_sm_secret_groups.existing_secret_groups.secret_groups[*].name, secret_group.secret_group_name)].id : null
+        secret_group_name = secret_group.secret_group_name != null ? secret_group.secret_group_name : "default"
```

Comment:
This secret_group_name appears redundant. It is included in a local that is only used to generate the secrets. Only the ID is passed to the secret creation.
The value ends up in the secrets output as part of the map name. This would mean that secrets had to be referenced by <group>.<name>, rather than just the secret name. (Note: secret names are unique per instance, and thus is sufficient to make a unique entry in the map).